### ISSUE TITLE
Different Behavior during ASPNET Core upgrade (MSI install vs Windows Update install)

### ISSUE THREAD

**Issue by RomBrz** [dotnet/aspnetcore#41377]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Last week i discovered that i could update the ASPNET Core versions on servers using WSUS updates and configure to deliver the ASPNET Core updates to help me managing 30+ servers.
Happen to the WSUS update doesn't make the server at same level comparing with a "Stand alone Host Bundle Install".

I have two servers "twins", with same ASPNET Core versions installed on both.
Server 1 i downloaded the Hosting Bundle packages (https://dotnet.microsoft.com/en-us/download/dotnet)
Server 2 i installed all the Windows Updates available (WSUS configured using https://devblogs.microsoft.com/dotnet/net-core-updates-coming-to-microsoft-update/)

Server1 after installing manually:
.NET Core Runtimes:
  6.0.4   x86
  6.0.4   x64
  5.0.16  x86
  5.0.16  x64
  3.1.24  x86
  3.1.24  x64

ASP.NET Core Runtimes:
  6.0.4   x86
  6.0.4   x64
  5.0.16  x86
  5.0.16  x64
  3.1.24  x86
  3.1.24  x64

.NET Core Runtime & Hosting Bundles:
  3.1.24
  6.0.4
  5.0.16

Server2 after WSUS install:
.NET Core Runtimes:
  6.0.3   x86
  6.0.3   x64
  5.0.16  x86
  5.0.16  x64
  3.1.24  x86
  3.1.24  x64

ASP.NET Core Runtimes:
  6.0.3   x86
  6.0.3   x64
  5.0.16  x86
  5.0.16  x64
  3.1.24  x86
  3.1.24  x64

.NET Core Runtime & Hosting Bundles:
  3.1.23
  6.0.4
  5.0.15

Seems like either the Hosting Bundle are update or the Runtime are updates.
3.1.24 and 5.0.16 updated the runtimes, but not the Hosting Bundle
6.0.4 updated the Hosting Bundle, but not the runtimes.

After reboot, and checking with Windows Update again, no updates are found.

### Expected Behavior

The WSUS/Windows Update of the ASPNET Core should deliver the same "updated version" that the intall of the stand alone Hosting Bundle package.

### .NET Version

all versions

**Comment by wtgodbe** [comment]
@jamshedd do we know what version(s) of the hosting bundle are now available on WSUS?

I just confirmed that installing the 6.0.3 Hosting Bundle followed by the 6.0.4 one will leave the 6.0.4 runtimes on your machine as expected, so I'm surprised to see the WSUS machine w/ the 6.0.4 Hosting Bundle but 6.0.3 runtimes

**Comment by jamshedd** [comment]
The removal of the older 6.0.3 runtime should not have anything to do with whether the hosting bundle was installed via WSUS or manually. Removal is of the older installers is handled by our bundle and the same bundle is executed in both cases. The removal of hte older runtime (or lack thereof) is likely a function of what else is on the machine (for e.g. VS) that might be referencing the older installer, or what app or service might have been running and locked files. 

Adding @csnellgrove1-zz to confirm.

**Comment by RomBrz** [comment]
I did more tests after this post.
Explaining better:
Server1: Installed manually the hosting bundle
Server2: Installed with WSUS -> Installing the 04/2022 patches (Windows, DotNet Framework, DotNet Core) -> DotNet 6.0.4 installation FAILED
Server3: Installed with WSUS -> Installing the 04/2022 patches (Windows, DotNet Framework, DotNet Core) -> DotNet 6.0.4 installation OK

Seems like the Server2 has some problem installing all the patches together. After reboot, the patch didn't show (Windows Update says that everything is up to date). (And the information appears like the initial post) I tried manually installing the 6.0.4 Hosting Bundle but didn't work. After some "good" time trying to solve what's wrong, because no matter what i did, i could not install the 6.0.4 (but could install the 6.0.3 after removing everything about 6.0.*), (ever tried install 6.0.4 after reinstall 6.0.3 but didn't work either) i reinstalled the 6.0.3, checked for updates, Windows Update found the 6.0.4 and installed OK. So seems like that the crash during the first Windows Update install broke and lock the server to only accept the 6.0.4 with WU.

About the Server3, it received the same patches that Server2 received, but installed all patches without problem. The result was that the server ends with the same components that Server2 (after all i posted above).

Both servers updated The 3.1.x, 5.0.x and 6.0.x runtimes (and removed the olders), but both didn't updated the ".NET Core Runtime & Hosting Bundles" for 3.1.x and 5.0.x (just the 6.0.x).

So at the end i have this:
Server1 (manually):
3.1.24 Runtimes / 3.1.24 Hosting Bundle
5.0.16 Runtimes / 5.0.16 Hosting Bundle
6.0.4 Runtimes / 6.0.4 Hosting Bundle

Server2 and Server3 (with WSUS):
3.1.24 Runtimes / 3.1.23 Hosting Bundle
5.0.16 Runtimes / 5.0.15 Hosting Bundle
6.0.4 Runtimes / 6.0.4 Hosting Bundle

**Comment by wtgodbe** [comment]
Could be related to the VS/SDK mismatch from 6.0.4 (we inadvertently shipped mismatched runtimes in Visual Studio vs. the Dotnet SDK in 6.0.4) - do these machines have VS installed? You could try the steps described here: https://developercommunity.visualstudio.com/t/PackageId:MicrosoftAspNetCoreSharedFra/10015847#T-N10018891

**Comment by RomBrz** [comment]
These servers are for publishing applications, so they don't have VS installed, and i never installed the SDK on it.

Before i knew that i  could update DotNet using WSUS/WU, i just downloaded each month the Hosting Bundle and install it on my IIS Servers so the DEV team could publish DotNet Core apps with existing DotNet Framework apps.

**Comment by RomBrz** [comment]
Trying again this month and trying to documentate the things:
Server before updates:
![image](https://user-images.githubusercontent.com/24864710/168168399-c6d9c5d1-3050-44c3-855e-a6d48e921b32.png)
The updates available by Windows Update:
![image](https://user-images.githubusercontent.com/24864710/168168463-131a20fa-c187-40b6-a02c-5fd17ef1d27a.png)
Server after updates:
![image](https://user-images.githubusercontent.com/24864710/168171595-3fab9997-77c9-4949-a5a8-ded7676c6150.png)


Seems like the Hosting Bundle for 3.1.x and 5.0.x are not updating with the Windows Update; only with manual installing of Hosting Bundle.

Expected behavior (i downloaded both Hosting Bundle from MS site and installed manually):
![image](https://user-images.githubusercontent.com/24864710/168172231-9b412032-15f9-413c-8926-6d2b2ec8c19e.png)

**Comment by wtgodbe** [comment]
Ah, that might be a hint - we recently did work to provide the Hosting Bundle through Microsoft Update - @jamshedd @joeloff @csnellgrove1-zz does MU offer the hosting bundle for 6.0 only, or for 3.1 & 5.0 as well? The change that makes it possible to offer the hosting bundle is present all the way back to 3.1, but I'm not sure whether or not MU actually offers it that far back.

**Comment by jamshedd** [comment]
Its offered for all versions (not just 6.0).

**Comment by wtgodbe** [comment]
The 3.1/5.0 problem sounds like an issue w/ Windows Update - @jamshedd @csnellgrove1-zz who can we talk to to confirm if/why the hosting bundle isn't being offered for 3.1/5.0? Sounds like @RomBrz has consistently not been offered it for the past 3 releases.

As for 6.0, are you able to attach logs from the failed install? They should be in your %Temp% folder after the install.

**Comment by joeloff** [comment]
The hosting bundle's offering depends on detecting the relevant ANCM copy. If the the 3.1 and 5.0 installs did not include ANCM, it won't be offered - this is one of the reasons why I suggested we should actually do the full bundle detection, rather than looking at the MSIs.

**Comment by RomBrz** [comment]
> The 3.1/5.0 problem sounds like an issue w/ Windows Update - @jamshedd @csnellgrove1-zz who can we talk to to confirm if/why the hosting bundle isn't being offered for 3.1/5.0? Sounds like @RomBrz has consistently not been offered it for the past 3 releases.
> 
> As for 6.0, are you able to attach logs from the failed install? They should be in your %Temp% folder after the install.

`[19F8:1538][2022-06-09T10:45:36]i001: Burn v3.14.0.5722, Windows v10.0 (Build 17763: Service Pack 0), path: C:\Windows\Temp\{37D57BA9-249F-4495-B053-69BCADB42776}\.cr\dotnet-hosting-6.0.5-win.exe
[19F8:1538][2022-06-09T10:45:36]i000: Initializing string variable 'BundleNameShort' to value 'Microsoft .NET 6.0.5'
[19F8:1538][2022-06-09T10:45:36]i000: Initializing string variable 'BundleNameSub' to value 'Windows Server Hosting'
[19F8:1538][2022-06-09T10:45:36]i000: Initializing string variable 'InstallResetIISState' to value 'disable'
[19F8:1538][2022-06-09T10:45:36]i000: Initializing string variable 'InstallNoIISState' to value 'disable'
[19F8:1538][2022-06-09T10:45:36]i000: Initializing string variable 'ModifyResetIISState' to value 'disable'
[19F8:1538][2022-06-09T10:45:36]i000: Initializing string variable 'ModifyNoIISState' to value 'disable'
[19F8:1538][2022-06-09T10:45:36]i009: Command Line: '-burn.clean.room=C:\Users\xxxxxxxxxx\Desktop\dotnet-hosting-6.0.5-win.exe -burn.filehandle.attached=672 -burn.filehandle.self=728'
[19F8:1538][2022-06-09T10:45:36]i000: Setting string variable 'WixBundleOriginalSource' to value 'C:\Users\xxxxxxxxxx\Desktop\dotnet-hosting-6.0.5-win.exe'
[19F8:1538][2022-06-09T10:45:36]i000: Setting string variable 'WixBundleOriginalSourceFolder' to value 'C:\Users\xxxxxxxxxx\Desktop\'
[19F8:1538][2022-06-09T10:45:36]i000: Setting string variable 'WixBundleLog' to value 'C:\Users\xxxxxxxxxx\AppData\Local\Temp\dd_dd_DotNetCoreWinSvrHosting__20220609104536.log'
[19F8:1538][2022-06-09T10:45:36]i000: Setting string variable 'WixBundleManufacturer' to value 'Microsoft Corporation'
[19F8:1CE4][2022-06-09T10:45:36]i000: Setting numeric variable 'WixStdBALanguageId' to value 1033
[19F8:1CE4][2022-06-09T10:45:36]i000: Setting version variable 'WixBundleFileVersion' to value '6.0.5.22218'
[19F8:1538][2022-06-09T10:45:36]i100: Detect begin, 7 packages
[19F8:1538][2022-06-09T10:45:36]i052: Condition 'VersionNT64' evaluates to true.
[19F8:1538][2022-06-09T10:45:36]i000: Product or related product not found: {1CCFDC95-8DDF-4610-97C6-6C0825D9E5F7}
[19F8:1538][2022-06-09T10:45:36]i000: Setting version variable 'DotNetRedistLtsProductVersion_x64' to value '0.0.0.0'
[19F8:1538][2022-06-09T10:45:36]i052: Condition 'NOT VersionNT64' evaluates to false.
[19F8:1538][2022-06-09T10:45:36]i000: Setting string variable 'IISCoreWebEngineInstalled_x86' to value '1'
[19F8:1538][2022-06-09T10:45:36]i000: Setting string variable 'IISW3SVCInstalled_x86' to value '1'
[19F8:1538][2022-06-09T10:45:36]i000: Setting string variable 'IISCoreWebEngineInstalled_x64' to value '1'
[19F8:1538][2022-06-09T10:45:36]i000: Setting string variable 'IISW3SVCInstalled_x64' to value '1'
[19F8:1538][2022-06-09T10:45:36]i052: Condition 'NOT (((IISCoreWebEngineInstalled_x86=1) AND (IISW3SVCInstalled_x86=1)) OR ((IISCoreWebEngineInstalled_x64=1) AND (IISW3SVCInstalled_x64=1)))' evaluates to false.
[19F8:1538][2022-06-09T10:45:36]i052: Condition '((IISCoreWebEngineInstalled_x86=1) AND (IISW3SVCInstalled_x86=1)) OR ((IISCoreWebEngineInstalled_x64=1) AND (IISW3SVCInstalled_x64=1))' evaluates to true.
[19F8:1538][2022-06-09T10:45:36]i000: Setting string variable 'InstallResetIISState' to value '1'
[19F8:1538][2022-06-09T10:45:36]i052: Condition 'NOT (((IISCoreWebEngineInstalled_x86=1) AND (IISW3SVCInstalled_x86=1)) OR ((IISCoreWebEngineInstalled_x64=1) AND (IISW3SVCInstalled_x64=1)))' evaluates to false.
[19F8:1538][2022-06-09T10:45:36]i052: Condition '((IISCoreWebEngineInstalled_x86=1) AND (IISW3SVCInstalled_x86=1)) OR ((IISCoreWebEngineInstalled_x64=1) AND (IISW3SVCInstalled_x64=1))' evaluates to true.
[19F8:1538][2022-06-09T10:45:36]i000: Setting string variable 'ModifyResetIISState' to value '1'
[19F8:1538][2022-06-09T10:45:36]i052: Condition 'NOT (OPT_NO_ANCM OR OPT_NO_ANCM=0 OR OPT_NO_FTS OR OPT_NO_FTS=0 OR OPT_NO_LTS OR OPT_NO_LTS=0 OR OPT_NO_SHAREDFX OR OPT_NO_SHAREDFX=0 OR OPT_NO_RUNTIME OR OPT_NO_RUNTIME=0 OR OPT_NO_X86 OR OPT_NO_X86=0 OR OPT_NO_SHARED_CONFIG_CHECK OR OPT_NO_SHARED_CONFIG_CHECK=0)' evaluates to true.
[19F8:1538][2022-06-09T10:45:36]i000: Setting numeric variable 'OPT_NO_ANCM_Exists' to value 1
[19F8:1538][2022-06-09T10:45:36]i052: Condition 'NOT (OPT_NO_ANCM OR OPT_NO_ANCM=0 OR OPT_NO_FTS OR OPT_NO_FTS=0 OR OPT_NO_LTS OR OPT_NO_LTS=0 OR OPT_NO_SHAREDFX OR OPT_NO_SHAREDFX=0 OR OPT_NO_RUNTIME OR OPT_NO_RUNTIME=0 OR OPT_NO_X86 OR OPT_NO_X86=0 OR OPT_NO_SHARED_CONFIG_CHECK OR OPT_NO_SHARED_CONFIG_CHECK=0)' evaluates to true.
[19F8:1538][2022-06-09T10:45:36]i000: Setting numeric variable 'OPT_NO_FTS_Exists' to value 1
[19F8:1538][2022-06-09T10:45:36]i052: Condition 'NOT (OPT_NO_ANCM OR OPT_NO_ANCM=0 OR OPT_NO_FTS OR OPT_NO_FTS=0 OR OPT_NO_LTS OR OPT_NO_LTS=0 OR OPT_NO_SHAREDFX OR OPT_NO_SHAREDFX=0 OR OPT_NO_RUNTIME OR OPT_NO_RUNTIME=0 OR OPT_NO_X86 OR OPT_NO_X86=0 OR OPT_NO_SHARED_CONFIG_CHECK OR OPT_NO_SHARED_CONFIG_CHECK=0)' evaluates to true.
[19F8:1538][2022-06-09T10:45:36]i000: Setting numeric variable 'OPT_NO_LTS_Exists' to value 1
[19F8:1538][2022-06-09T10:45:36]i052: Condition 'NOT (OPT_NO_ANCM OR OPT_NO_ANCM=0 OR OPT_NO_FTS OR OPT_NO_FTS=0 OR OPT_NO_LTS OR OPT_NO_LTS=0 OR OPT_NO_SHAREDFX OR OPT_NO_SHAREDFX=0 OR OPT_NO_RUNTIME OR OPT_NO_RUNTIME=0 OR OPT_NO_X86 OR OPT_NO_X86=0 OR OPT_NO_SHARED_CONFIG_CHECK OR OPT_NO_SHARED_CONFIG_CHECK=0)' evaluates to true.
[19F8:1538][2022-06-09T10:45:36]i000: Setting numeric variable 'OPT_NO_RUNTIME_Exists' to value 1
[19F8:1538][2022-06-09T10:45:36]i052: Condition 'NOT (OPT_NO_ANCM OR OPT_NO_ANCM=0 OR OPT_NO_FTS OR OPT_NO_FTS=0 OR OPT_NO_LTS OR OPT_NO_LTS=0 OR OPT_NO_SHAREDFX OR OPT_NO_SHAREDFX=0 OR OPT_NO_RUNTIME OR OPT_NO_RUNTIME=0 OR OPT_NO_X86 OR OPT_NO_X86=0 OR OPT_NO_SHARED_CONFIG_CHECK OR OPT_NO_SHARED_CONFIG_CHECK=0)' evaluates to true.
[19F8:1538][2022-06-09T10:45:36]i000: Setting numeric variable 'OPT_NO_SHARED_CONFIG_CHECK_Exists' to value 1
[19F8:1538][2022-06-09T10:45:36]i052: Condition 'NOT (OPT_NO_ANCM OR OPT_NO_ANCM=0 OR OPT_NO_FTS OR OPT_NO_FTS=0 OR OPT_NO_LTS OR OPT_NO_LTS=0 OR OPT_NO_SHAREDFX OR OPT_NO_SHAREDFX=0 OR OPT_NO_RUNTIME OR OPT_NO_RUNTIME=0 OR OPT_NO_X86 OR OPT_NO_X86=0 OR OPT_NO_SHARED_CONFIG_CHECK OR OPT_NO_SHARED_CONFIG_CHECK=0)' evaluates to true.
[19F8:1538][2022-06-09T10:45:36]i000: Setting numeric variable 'OPT_NO_SHAREDFX_Exists' to value 1
[19F8:1538][2022-06-09T10:45:36]i052: Condition 'NOT (OPT_NO_ANCM OR OPT_NO_ANCM=0 OR OPT_NO_FTS OR OPT_NO_FTS=0 OR OPT_NO_LTS OR OPT_NO_LTS=0 OR OPT_NO_SHAREDFX OR OPT_NO_SHAREDFX=0 OR OPT_NO_RUNTIME OR OPT_NO_RUNTIME=0 OR OPT_NO_X86 OR OPT_NO_X86=0 OR OPT_NO_SHARED_CONFIG_CHECK OR OPT_NO_SHARED_CONFIG_CHECK=0)' evaluates to true.
[19F8:1538][2022-06-09T10:45:36]i000: Setting numeric variable 'OPT_NO_X86_Exists' to value 1
[19F8:1538][2022-06-09T10:45:36]i052: Condition 'VersionNT64' evaluates to true.
[19F8:1538][2022-06-09T10:45:36]i000: Setting version variable 'SharedFxRedistProductVersion_x64' to value '6.0.5.22218'
[19F8:1538][2022-06-09T10:45:36]i052: Condition 'NOT VersionNT64' evaluates to false.
[19F8:1538][2022-06-09T10:45:36]i052: Condition 'OPT_NO_SHAREDFX_Exists' evaluates to true.
[19F8:1538][2022-06-09T10:45:36]i000: Setting string variable 'OPT_NO_SHAREDFX' to value '0'
[19F8:1538][2022-06-09T10:45:36]i052: Condition 'OPT_NO_FTS_Exists' evaluates to true.
[19F8:1538][2022-06-09T10:45:36]i000: Setting string variable 'OPT_NO_FTS' to value '0'
[19F8:1538][2022-06-09T10:45:36]i052: Condition 'OPT_NO_X86_Exists' evaluates to true.
[19F8:1538][2022-06-09T10:45:36]i000: Setting string variable 'OPT_NO_X86' to value '0'
[19F8:1538][2022-06-09T10:45:36]i052: Condition 'OPT_NO_LTS_Exists' evaluates to true.
[19F8:1538][2022-06-09T10:45:36]i000: Setting string variable 'OPT_NO_LTS' to value '0'
[19F8:1538][2022-06-09T10:45:36]i052: Condition 'OPT_NO_RUNTIME_Exists' evaluates to true.
[19F8:1538][2022-06-09T10:45:36]i000: Setting string variable 'OPT_NO_RUNTIME' to value '0'
[19F8:1538][2022-06-09T10:45:36]i052: Condition 'OPT_NO_SHARED_CONFIG_CHECK_Exists' evaluates to true.
[19F8:1538][2022-06-09T10:45:36]i000: Setting string variable 'OPT_NO_SHARED_CONFIG_CHECK' to value '0'
[19F8:1538][2022-06-09T10:45:36]i052: Condition 'OPT_NO_ANCM_Exists' evaluates to true.
[19F8:1538][2022-06-09T10:45:36]i000: Setting string variable 'OPT_NO_ANCM' to value '0'
[19F8:1538][2022-06-09T10:45:36]i052: Condition 'DotNetRedistLtsProductVersion_x64 = v48.23.40665' evaluates to false.
[19F8:1538][2022-06-09T10:45:36]i052: Condition 'DotNetRedistLtsProductVersion_x86 = v48.23.40665' evaluates to false.
[19F8:1538][2022-06-09T10:45:36]i052: Condition 'SharedFxRedistProductVersion_x64 = v6.0.5.22218' evaluates to true.
[19F8:1538][2022-06-09T10:45:36]i052: Condition 'SharedFxRedistProductVersion_x86 = v6.0.5.22218' evaluates to false.
[19F8:1538][2022-06-09T10:45:36]i101: Detected package: WindowsServerHostingBundleOptions.msi, state: Present, cached: Complete
[19F8:1538][2022-06-09T10:45:36]i101: Detected package: AspNetCoreModuleV2_x86, state: Absent, cached: None
[19F8:1538][2022-06-09T10:45:36]i101: Detected package: AspNetCoreModuleV2_x64, state: Present, cached: Complete
[19F8:1538][2022-06-09T10:45:36]i101: Detected package: DotNetRedistLts_x64, state: Absent, cached: Complete
[19F8:1538][2022-06-09T10:45:36]i101: Detected package: DotNetRedistLts_x86, state: Absent, cached: Complete
[19F8:1538][2022-06-09T10:45:36]i101: Detected package: SharedFxRedist_x64, state: Present, cached: Complete
[19F8:1538][2022-06-09T10:45:36]i101: Detected package: SharedFxRedist_x86, state: Absent, cached: Complete
[19F8:1538][2022-06-09T10:45:36]i199: Detect complete, result: 0x0
[19F8:1CE4][2022-06-09T10:45:36]i000: Disable control ModifyNoIIS
[19F8:1538][2022-06-09T10:45:37]i200: Plan begin, 7 packages, action: Repair
[19F8:1538][2022-06-09T10:45:37]i000: Setting string variable 'WixBundleLog_WindowsServerHostingBundleOptions.msi' to value 'C:\Users\xxxxxxxxxx~1.FAR\AppData\Local\Temp\dd_dd_DotNetCoreWinSvrHosting__20220609104536_000_WindowsServerHostingBundleOptions.msi.log'
[19F8:1538][2022-06-09T10:45:37]i052: Condition 'NOT VersionNT64 AND (VersionNT >= v6.1) AND (IISCoreWebEngineInstalled_x86=1) AND (IISW3SVCInstalled_x86=1) AND (NOT OPT_NO_ANCM OR OPT_NO_ANCM="0")' evaluates to false.
[19F8:1538][2022-06-09T10:45:37]i052: Condition 'VersionNT64 AND (VersionNT64 >= v6.1) AND (IISCoreWebEngineInstalled_x64=1) AND (IISW3SVCInstalled_x64=1) AND (NOT OPT_NO_ANCM OR OPT_NO_ANCM="0")' evaluates to true.
[19F8:1538][2022-06-09T10:45:37]i000: Setting string variable 'WixBundleLog_AspNetCoreModuleV2_x64' to value 'C:\Users\xxxxxxxxxx~1.FAR\AppData\Local\Temp\dd_dd_DotNetCoreWinSvrHosting__20220609104536_001_AspNetCoreModuleV2_x64.log'
[19F8:1538][2022-06-09T10:45:37]i052: Condition 'VersionNT64 AND (NOT OPT_NO_RUNTIME OR OPT_NO_RUNTIME="0")' evaluates to true.
[19F8:1538][2022-06-09T10:45:37]w321: Skipping dependency registration on package with no dependency providers: DotNetRedistLts_x64
[19F8:1538][2022-06-09T10:45:37]i000: Setting string variable 'WixBundleLog_DotNetRedistLts_x64' to value 'C:\Users\xxxxxxxxxx~1.FAR\AppData\Local\Temp\dd_dd_DotNetCoreWinSvrHosting__20220609104536_002_DotNetRedistLts_x64.log'
[19F8:1538][2022-06-09T10:45:37]i000: Setting string variable 'WixBundleRollbackLog_DotNetRedistLts_x64' to value 'C:\Users\xxxxxxxxxx~1.FAR\AppData\Local\Temp\dd_dd_DotNetCoreWinSvrHosting__20220609104536_002_DotNetRedistLts_x64_rollback.log'
[19F8:1538][2022-06-09T10:45:37]i052: Condition '(NOT OPT_NO_RUNTIME OR OPT_NO_RUNTIME="0") AND (NOT OPT_NO_X86 OR OPT_NO_X86="0")' evaluates to true.
[19F8:1538][2022-06-09T10:45:37]w321: Skipping dependency registration on package with no dependency providers: DotNetRedistLts_x86
[19F8:1538][2022-06-09T10:45:37]i000: Setting string variable 'WixBundleLog_DotNetRedistLts_x86' to value 'C:\Users\xxxxxxxxxx~1.FAR\AppData\Local\Temp\dd_dd_DotNetCoreWinSvrHosting__20220609104536_003_DotNetRedistLts_x86.log'
[19F8:1538][2022-06-09T10:45:37]i000: Setting string variable 'WixBundleRollbackLog_DotNetRedistLts_x86' to value 'C:\Users\xxxxxxxxxx~1.FAR\AppData\Local\Temp\dd_dd_DotNetCoreWinSvrHosting__20220609104536_003_DotNetRedistLts_x86_rollback.log'
[19F8:1538][2022-06-09T10:45:37]i052: Condition 'VersionNT64 AND (NOT OPT_NO_SHAREDFX OR OPT_NO_SHAREDFX="0")' evaluates to true.
[19F8:1538][2022-06-09T10:45:37]w321: Skipping dependency registration on package with no dependency providers: SharedFxRedist_x64
[19F8:1538][2022-06-09T10:45:37]i000: Setting string variable 'WixBundleLog_SharedFxRedist_x64' to value 'C:\Users\xxxxxxxxxx~1.FAR\AppData\Local\Temp\dd_dd_DotNetCoreWinSvrHosting__20220609104536_004_SharedFxRedist_x64.log'
[19F8:1538][2022-06-09T10:45:37]i052: Condition '(NOT OPT_NO_SHAREDFX OR OPT_NO_SHAREDFX="0") AND (NOT OPT_NO_X86 OR OPT_NO_X86="0")' evaluates to true.
[19F8:1538][2022-06-09T10:45:37]w321: Skipping dependency registration on package with no dependency providers: SharedFxRedist_x86
[19F8:1538][2022-06-09T10:45:37]i000: Setting string variable 'WixBundleLog_SharedFxRedist_x86' to value 'C:\Users\xxxxxxxxxx~1.FAR\AppData\Local\Temp\dd_dd_DotNetCoreWinSvrHosting__20220609104536_005_SharedFxRedist_x86.log'
[19F8:1538][2022-06-09T10:45:37]i000: Setting string variable 'WixBundleRollbackLog_SharedFxRedist_x86' to value 'C:\Users\xxxxxxxxxx~1.FAR\AppData\Local\Temp\dd_dd_DotNetCoreWinSvrHosting__20220609104536_005_SharedFxRedist_x86_rollback.log'
[19F8:1538][2022-06-09T10:45:37]i201: Planned package: WindowsServerHostingBundleOptions.msi, state: Present, default requested: Repair, ba requested: Repair, execute: Repair, rollback: None, cache: No, uncache: No, dependency: Register
[19F8:1538][2022-06-09T10:45:37]i201: Planned package: AspNetCoreModuleV2_x86, state: Absent, default requested: Absent, ba requested: Absent, execute: None, rollback: None, cache: No, uncache: No, dependency: None
[19F8:1538][2022-06-09T10:45:37]i201: Planned package: AspNetCoreModuleV2_x64, state: Present, default requested: Repair, ba requested: Repair, execute: Repair, rollback: None, cache: No, uncache: No, dependency: Register
[19F8:1538][2022-06-09T10:45:37]i201: Planned package: DotNetRedistLts_x64, state: Absent, default requested: Repair, ba requested: Repair, execute: Install, rollback: Uninstall, cache: No, uncache: No, dependency: None
[19F8:1538][2022-06-09T10:45:37]i201: Planned package: DotNetRedistLts_x86, state: Absent, default requested: Repair, ba requested: Repair, execute: Install, rollback: Uninstall, cache: No, uncache: No, dependency: None
[19F8:1538][2022-06-09T10:45:37]i201: Planned package: SharedFxRedist_x64, state: Present, default requested: Repair, ba requested: Repair, execute: Repair, rollback: None, cache: No, uncache: No, dependency: None
[19F8:1538][2022-06-09T10:45:37]i201: Planned package: SharedFxRedist_x86, state: Absent, default requested: Repair, ba requested: Repair, execute: Install, rollback: Uninstall, cache: No, uncache: No, dependency: None
[19F8:1538][2022-06-09T10:45:37]i299: Plan complete, result: 0x0
[19F8:1538][2022-06-09T10:45:37]i300: Apply begin
[19F8:1538][2022-06-09T10:45:37]i010: Launching elevated engine process.
[19F8:1538][2022-06-09T10:45:37]i011: Launched elevated engine process.
[19F8:1538][2022-06-09T10:45:37]i012: Connected to elevated engine.
[0910:06C8][2022-06-09T10:45:37]i358: Pausing automatic updates.
[0910:06C8][2022-06-09T10:45:37]i359: Paused automatic updates.
[0910:06C8][2022-06-09T10:45:37]i360: Creating a system restore point.
[0910:06C8][2022-06-09T10:45:37]i362: System restore disabled, system restore point not created.
[0910:06C8][2022-06-09T10:45:37]i371: Updating session, registration key: SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{623bc5ad-1ff0-4d68-95ee-02a5d4c10bf6}, resume: Active, restart initiated: No, disable resume: No
[0910:06E4][2022-06-09T10:45:37]i304: Verified existing payload: WindowsServerHostingBundleOptions.msi at path: C:\ProgramData\Package Cache\{A2C87A3D-ACEB-34B6-8BC3-055E1055A118}v6.0.5.22218\WindowsServerHostingBundleOptions.msi.
[0910:06C8][2022-06-09T10:45:37]i323: Registering package dependency provider: {A2C87A3D-ACEB-34B6-8BC3-055E1055A118}, version: 6.0.5.22218, package: WindowsServerHostingBundleOptions.msi
[0910:06C8][2022-06-09T10:45:37]i301: Applying execute package: WindowsServerHostingBundleOptions.msi, action: Repair, path: C:\ProgramData\Package Cache\{A2C87A3D-ACEB-34B6-8BC3-055E1055A118}v6.0.5.22218\WindowsServerHostingBundleOptions.msi, arguments: ' ARPSYSTEMCOMPONENT="1" MSIFASTINSTALL="7" OPT_NO_ANCM="0" OPT_NO_FTS="0" OPT_NO_LTS="0" OPT_NO_SHAREDFX="0" OPT_NO_RUNTIME="0" OPT_NO_X86="0" OPT_NO_SHARED_CONFIG_CHECK="0"'
[0910:06E4][2022-06-09T10:45:37]i304: Verified existing payload: AspNetCoreModuleV2_x64 at path: C:\ProgramData\Package Cache\{93FF8269-10C8-4C2F-8D9F-23AD420B6543}v16.0.22108.0\AspNetCoreModuleV2_x64.msi.
[0910:06E4][2022-06-09T10:45:37]i304: Verified existing payload: DotNetRedistLts_x64 at path: C:\ProgramData\Package Cache\56C1D5B7DCC6F34DF779BFEC38610C7353324E76\dotnet-runtime-6.0.5-win-x64.exe.
[19F8:1538][2022-06-09T10:45:37]i319: Applied execute package: WindowsServerHostingBundleOptions.msi, result: 0x0, restart: None
[0910:06C8][2022-06-09T10:45:37]i325: Registering dependency: {623bc5ad-1ff0-4d68-95ee-02a5d4c10bf6} on package provider: {A2C87A3D-ACEB-34B6-8BC3-055E1055A118}, package: WindowsServerHostingBundleOptions.msi
[0910:06C8][2022-06-09T10:45:37]i301: Applying execute package: AspNetCoreModuleV2_x64, action: Repair, path: C:\ProgramData\Package Cache\{93FF8269-10C8-4C2F-8D9F-23AD420B6543}v16.0.22108.0\AspNetCoreModuleV2_x64.msi, arguments: ' ARPSYSTEMCOMPONENT="1" MSIFASTINSTALL="7" OPT_NO_SHARED_CONFIG_CHECK="0"'
[0910:06E4][2022-06-09T10:45:37]i304: Verified existing payload: DotNetRedistLts_x86 at path: C:\ProgramData\Package Cache\0FCD8B3E79048725B033CF965DEEE2079911856D\dotnet-runtime-6.0.5-win-x86.exe.
[0910:06E4][2022-06-09T10:45:38]i304: Verified existing payload: SharedFxRedist_x64 at path: C:\ProgramData\Package Cache\08AB9FF5AF15C6A3F8630ED598050CF000CD1F7A\aspnetcore-runtime-6.0.5-win-x64.exe.
[0910:06E4][2022-06-09T10:45:38]i304: Verified existing payload: SharedFxRedist_x86 at path: C:\ProgramData\Package Cache\D45393A8141A322BED6CC5685BB807D1BFBC06C4\aspnetcore-runtime-6.0.5-win-x86.exe.
[0910:06C8][2022-06-09T10:45:40]e000: Error 0x80070643: Failed to run maintanance mode for MSI package.
[0910:06C8][2022-06-09T10:45:40]e000: Error 0x80070643: Failed to execute MSI package.
[19F8:1538][2022-06-09T10:45:40]e000: Error 0x80070643: Failed to configure per-machine MSI package.
[19F8:1538][2022-06-09T10:45:40]i319: Applied execute package: AspNetCoreModuleV2_x64, result: 0x80070643, restart: None
[19F8:1538][2022-06-09T10:45:40]e000: Error 0x80070643: Failed to execute MSI package.
[0910:06C8][2022-06-09T10:45:40]i372: Session end, registration key: SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{623bc5ad-1ff0-4d68-95ee-02a5d4c10bf6}, resume: ARP, restart: None, disable resume: No
[0910:06C8][2022-06-09T10:45:40]i371: Updating session, registration key: SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{623bc5ad-1ff0-4d68-95ee-02a5d4c10bf6}, resume: ARP, restart initiated: No, disable resume: No
[19F8:1538][2022-06-09T10:45:40]i399: Apply complete, result: 0x80070643, restart: None, ba requested restart:  No
`

**Comment by RomBrz** [comment]
> Can you provide us with the Windows Update log? Run `Get-WindowsUpdateLog` from PowerShell. Also, can you confirm if any of these machines had VS installed?

The last patch of servers that i updated didn't have de problem.
Today i have started the 06/2022 patches.

Just 3.1.x and 6.0.x versions to update using Windows Update.
3.1.26 updated the Runtime, didn't updated the Server Bundled (manually installing after Windows Update updates everything);
6.0.6 updated both, and my validation is run the Server Bundled exe and see if the installation package will ask me to "Reinstall", and the package needs to finish OK.
So seems like the 6.0.6 didn't crash.

**Comment by RomBrz** [comment]
Another month, the problems persists.
3.1.27 still doesn't update the Hosting Bundle;

About DotNet 6, seems like sometimes it crashes when the IIS is running. Last month and this month i tried another approach: before start the Windows Update installation, i changed the IIS Service startup to "Manual", rebooted the server, and then i install the updates (windows + dotnet). This way the installation is working fine.
All the crashes that i have was when i didn't stop the IIS Service (just remove from the load balance and apply the patches), with some DotNet6 applicatons running. Maybe IIS is locking some DLLs that was updated when the installation by the Windows Update?
I don't know exactly why, but installing the Hosting Bundle MSI i never saw that problem.

To be sure, since i enabled the WSUS update of DotNet, when the Windows Update installs the patch i pick up the Hosting Bundle MSI and do a "repair" to force the reinstall and check if it was installed OK: when the intalls have a bug, the "Repair" from the MSI didn't work, so i have to do what i already told.

When crash i saw that the Hosting Bundle was updated but the runtimes not. Maybe this helps.

**Comment by joeloff** [comment]
@RomBrz you have both 3.1 and 6.0 hosting bundles installed?

**Comment by RomBrz** [comment]
Yes, both installed. When the 6.0.x doesn't crash (avoided stopping IIS
before starting the installation) the WSUS versiom update both runtimes and
hosting bundle (same behavior that the MSI package).
And also seems to respect what componet is installed (some servers has only
the Hosting Bundle, not the runtimes).

Em ter., 26 de jul. de 2022 19:02, Jacques Eloff ***@***.***>
escreveu:

**Comment by RomBrz** [comment]
@joeloff 
Seems like the 3.1.x is solved (both Hosting Bundle and DotNet been updated).
The bug with the 6.0 persists: If the IIS is running an DotNet 6 application the Windows Update crashes and not install. The fix is:
1) change IIS to manual;
2) reboot the server;
3) apply the patches (windows + dotnet);
4) reboot to check if there's some update not installed;
5) change IIS to automatic;
6) reboot the server.

In this way the DotNet 6 updates OK with Windows Update.

**Comment by RomBrz** [comment]
I have a server with just "Hosting Bundle" installed (not the runtimes), this month patch updates the actual Hosting Bundle for both DotNet (3.1.x and 6.0.x) but install the runtimes.
In my opinion, the update should only update what is already installed, not install the full pack and update everything else..

**Comment by wtgodbe** [comment]
Do you have the update logs? The hosting bundle is designed to only update what is already installed, if it updated everything that sounds like a bug.

**Comment by RomBrz** [comment]
I'm sending 3 logs:
1) WindowsUpdate-DotNet6Failed.txt = Server Failed to install DotNet 6.0.x while IIS is running.
[WindowsUpdate-DotNet6Failed.txt](https://github.com/dotnet/aspnetcore/files/9433436/WindowsUpdate-DotNet6Failed.txt)

2) WindowsUpdate-DotNet3Runtime.txt = Server has only Hosting Bundle for 3.1.x, after the update, has Runtime too.
[WindowsUpdate-DotNet3Runtime.txt](https://github.com/dotnet/aspnetcore/files/9433435/WindowsUpdate-DotNet3Runtime.txt)

3) WindowsUpdate-DotNet6Runtime.txt = Same as 2, but with DotNet 6.0.x.
[WindowsUpdate-DotNet6Runtime.txt](https://github.com/dotnet/aspnetcore/files/9433437/WindowsUpdate-DotNet6Runtime.txt)

**Comment by wtgodbe** [comment]
@RomBrz we would need the MSI logs to determine what went wrong - they'd be in your %temp% folder (of the user that ran the installation), search for %temp%\dd*hosting.log, they should look similar to dd_dd_DotNetCoreWinSvrHosting__20220819002058.log

**Comment by RomBrz** [comment]
1. The log from the server with IIS running that crashed trying to install DotNet 6.
[dd_dd_DotNetCoreWinSvrHosting__20220818090804.log](https://github.com/dotnet/aspnetcore/files/9436289/dd_dd_DotNetCoreWinSvrHosting__20220818090804.log)

2. The server that have only the Hosting Bundle installed. It doesn't have the "dd_dd_*" files, only 4 types of files (3.1.x both x86/x64 and 6.0.x both x86/x64), so i'm sending just one of then the 6.0.x x64.
[Microsoft_.NET_Runtime_-_6.0.8_(x64)_20220825121103.log](https://github.com/dotnet/aspnetcore/files/9436288/Microsoft_.NET_Runtime_-_6.0.8_.x64._20220825121103.log)

**Comment by wtgodbe** [comment]
For the first, you're hitting Error 0x80070643, which is often caused by a corrupted .Net Framework on the machine - you could try repairing .Net Framework before re-running the install.

For the second, is there a log with DotNetCoreWinSvrHosting in the name on the machine? That'd have what we need to figure out the problem

**Comment by RomBrz** [comment]
> For the first, you're hitting Error 0x80070643, which is often caused by a corrupted .Net Framework on the machine - you could try repairing .Net Framework before re-running the install.

As i told before, i can reproduce this error if i click on "Install Windows Updates" while IIS is running; the DotNet 3.1 installs OK while IIS running, DotNet 6 gives this crash and corrupt the installation. In this case: Tried to install, gives this bug; i have to manually uninstall everything related to 6.0.8 AND 6.0.7 (some packages keeps both versions), on this server, change IIS startup to manual/disabled, reboot the server, reinstall the 6.0.7 with the Hosting Bundle Package, THAN install manually the 6.0.8 to work.
I can reproduce this, have two more servers to update with the same config: If i keep IIS running, will crash. If i disable IIS before setup, installs OK.

> For the second, is there a log with DotNetCoreWinSvrHosting in the name on the machine? That'd have what we need to figure out the problem

Doesn't have this. All files from the aug 25 (yesterday):
![Logs Available](https://user-images.githubusercontent.com/24864710/186994137-f7f817df-65e2-40fb-bf3e-e5f330b7473b.png)

**Comment by RomBrz** [comment]
Any update?

**Comment by wtgodbe** [comment]
Sorry for the delays, I've been busy as we're finishing up work for 7.0. I'll work on getting a repro of the 6.0 failure in the near future.

**Comment by RomBrz** [comment]
Installing 6.0.9 and 3.1.29.
Tried this month without rebooting and/or stopping IIS before install the patches.
3.1.29 installed OK.
6.0.9 give an error, i clicked on retry and seems to install. But after reboot, seems like the Hosting Bundle 6.0.8 wasn't uninstalled:
![DotNetCore](https://user-images.githubusercontent.com/24864710/191250226-3af8785e-7182-4236-ac5e-547c5fb2d020.png)

Looking at this, seems like that the WU installation of the DotNet 6.x crashes if there's any app running at DotNet 6.x on IIS, and seems like it's related to the Hosting Bundle package itself (as the Runtimes and Shared Framework are both updated).

This not happens if i change IIS startup to manual and reboot the server before start the WU installation.

**Comment by RomBrz** [comment]
Any update? Right now i'm stopping IIS before WU installation and seems to work.
But the problem to install everything instead of just update what is already installed continues.
The server with only Windows Server Hosting + Shared files (not the runtime) after WU installation get to update both Server Hosting + Shared Files AND the latest runtime.
I have to run the dotnet-uninstall to remove all the runtimes (in the case, 3.1 runtime x64 and x86 + 6.0 runtime x64 and x86).

For me the behavior should be just update what is installed

**Comment by RomBrz** [comment]
Updating the thread, seems like that with updating DotNet 6 and 7 the problem doesn't happen. One time happened (last month) but the Windows Update itselt tried again and updated okay.
As 3.1 and 5.0 is on EOL status, the updates aren't a problem anymore.

The only problem i have right now it's that the WU Installer didn't get what i have installed, so in my case (i install the full hosting bundle when new DotNet is release than uninstall the runtimes, keeping only the Hosting Bundle package), the servers that doesn't have the Runtimes keep installing the Runtimes after WU updates. 

But seems like a different bug, so if this persist i will open another issue.

Thanks
